### PR TITLE
Added sell multipliers

### LINKED ISSUE #3934: /sell multipliers with permissions
### Feature request

**Feature description:**  

/sell multipliers that can be set with permission nodes.
The multipliers are to increase/multiply the sell price that someone can get via the /sell command. It basically multiplies the worth.

Example permissions:
essentials.sell.multiplier.0.5 > multiplies the worth/sell price by 0.5
essentials.sell.multiplier.2 > multiplies the worth/sell price by 2
essentials.sell.multiplier.X > multiplies the worth/sell price by X
OR
essentials.worth.multiplier.0.5 > multiplies the worth/sell price by 0.5
essentials.worth.multiplier.2 > multiplies the worth/sell price by 2
essentials.worth.multiplier.X > multiplies the worth/sell price by X

**How the feature is useful:**  
It is useful as it allows you to give a certain benefit to certain groups whilst still taking advantage of the EssentialsX worth and /sell features.
Its particular useful to server owners as it means they do not need to search and potentially purchase a separate plugin for such a small addition. You would have to re-create the shops and all their prices again.
The feature is player friendly as they can still use one command, rather than needing to visit different sign shop areas, or using different commands to access different shops that have different prices.

**Note**
I filtered through the issues (open and closed) and also checked changelogs on the GitHub to make sure I was not making a duplicate and/or denied request. My apologises if I missed one.

### PR DESCRIPTION
**Pull request by justinmtech**
### Information

This PR closes #3934

### Details

**Proposed feature:**  

Sell multipliers! The idea is for servers to be able to control how much players can /sell their items for based on configurable permissions. I re-wrote this based on @Chew's feedback to work more similarly to the multiple set-home system so that it doesn't have to iterate over all of a players' permissions and instead, check if they have the ones in the config.
https://github.com/EssentialsX/Essentials/issues/3934

**Environments tested:**    

OS: Windows 10

Java version:  OpenJDK 17.0.1

- [x] Most recent Paper version (1.18.1, #153)
- [ ] CraftBukkit/Spigot/Paper 1.12.2
- [ ] CraftBukkit 1.8.8

**Demonstration:**  

Screenshot: [https://i.imgur.com/ju0jMUy.png](url)
Here I give myself a diamond and sell it with no permissions (as op), then I give myself the "essentials.sell.multiplier.double" permission and as you can see, my sell price is double.
Plugin Download: https://www.dropbox.com/s/56y3zf6ibarlwj4/EssentialsX-2.19.3-dev%2B6-739600e-sell-multipliers-redo.jar?dl=0

### CODE REVIEW

File: Essentials/src/main/java/com/earth2me/essentials/ISettings.java [hunk 1]
```diff
@@ -405,6 +405,7 @@ public interface ISettings extends IConf {
     boolean isUpdateCheckEnabled();
 
     boolean showZeroBaltop();
+    BigDecimal getMultiplier(final User user);
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/ISettings.java:421]:
```suggestion

    BigDecimal getMultiplier(final User user);
```

File: Essentials/src/main/java/com/earth2me/essentials/commands/Commandsell.java [hunk 2]
```diff
@@ -91,7 +91,7 @@ public void run(final Server server, final User user, final String commandLabel,
 
     private BigDecimal sellItem(final User user, final ItemStack is, final String[] args, final boolean isBulkSell) throws Exception {
         final int amount = ess.getWorth().getAmount(ess, user, is, args, isBulkSell);
-        final BigDecimal worth = ess.getWorth().getPrice(ess, is);
+        final BigDecimal worth = ess.getWorth().getPrice(ess, is) == null ? null : ess.getWorth().getPrice(ess, is).multiply(ess.getSettings().getMultiplier(user));
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/commands/Commandsell.java]:
```suggestion
        final BigDecimal originalWorth = ess.getWorth().getPrice(ess, is);
        final BigDecimal worth = originalWorth == null ? null : originalWorth.multiply(ess.getSettings().getMultiplier(user));
```

File: Essentials/src/main/resources/config.yml [hunk 3]
```diff
@@ -854,6 +854,17 @@ show-zero-baltop: true
 # For 1'234,50 use fr-ch
 #currency-symbol-format-locale: en-US
 
+# Settings for /sell multipliers
+# Give multipliers to players with the permission 'essentials.sell.multiplier.{multiplier}'
+# example: 'essentials.sell.multiplier.double', 'essentials.sell.multiplier.triple'
+# You can add as many multipliers as you want.
+# Use the "default" name to set the default multiplier that everyone has. It defaults to 1 if not set.
+# If a player has more than one multiplier permission, it will pick the highest one.
```
Comment (JRoy) [inline Essentials/src/main/resources/config.yml]:
```suggestion
# Allow players to receive multipliers for items sold with /sell or the sell sign.
# You can set the default multiplier using the 'default' rank below.
# To grant different multipliers to different people, you need to define a 'multiplier-rank' below.
# Create the 'multiplier-rank' below, and give the matching permission: essentials.sell.multiplier.<multiplier-rank>
```

**Review by JRoy — CHANGES_REQUESTED** [review]
also needs to account for sell sign

**Comment by zzkanezz2** [comment]
Can we please get sell multipliers implemented to latest essentialsx?

**Review by JRoy — APPROVED** [review]

**Pull request merged**